### DISCUSSION TITLE
Getting ALE/isort v5 to play nicely

### DISCUSSION THREAD

**Discussion by Gidgidonihah** [w0rp/ale#3794] [Q&A - Ask for help with problems] [answered]
I've been using ALE for a good long time. As of isort v5, they no longer play nicely together for me. And, a year after the release, I've finally dug into what's going on to try to fix it.

Given the following project structure:
```
tree isort-test
isort-test
├── main.py
├── mod1
│   └── foo.py
└── mod2
    └── bar
        └── baz.py
```

where `mod2/bar/baz.py`'s imports look like:
```py
import os
import requests
from mod1 import foo
```

WIth v4 you could run this and end up with imports that look like:
```sh
cd isort-test/mod2/bar 
isort baz.py
```

```py
import os

import requests

from mod1 import foo
```

With v5 if you run the same command, it doesn't know what the local imports are and you get:
```sh
cd isort-test/mod2/bar 
isort baz.py
```

```py
import os

import requests
from mod1 import foo
```

No problem. That's what the `--src` command is for:
```sh
cd isort-test/mod2/bar 
isort --src=../.. baz.py
```

```py
import os

import requests

from mod1 import foo
```

Or alternatively, always run isort from the project root:
```sh
cd isort-test/
isort mode2/bar/baz.py
```
```py
import os

import requests

from mod1 import foo
```

However, I don't see a way to configure that for ALE.

1. I can't set my source root in my project `pyproject.toml` or `isort.cfg` file because I use one global config (consumed via CI) file for all the many repos that I may be working with.
2. I can't run from the project root (assuming I started vim from the project root) because the isort fixer [changes the CWD](https://github.com/dense-analysis/ale/blob/f0887d3e6178482255f11aa378124aef3699245f/autoload/ale/fixers/isort.vim#L30) to the dir of the file.

The only option I can think of without modifying the fixer would be something like:
```vim
let g:ale_python_isort_options='--settings-path=/Users/myself/pyproject.toml --src {something here for buffer cwd that doesn't respect the cwd change the fixer makes}'
```
I've tried various combinations of `%:p:h` to no avail, and my gut says even if that did work, it would get expanded after the change of directory.

What simple config am I missing? Or alternatively, does the fixer need to change to make the CWD change optional?

I've gotten around this temporarily by manually modifying the fixer to not set the cwd in my local plugin, and adding `--src .` to the `ale_python_isort_options`, but that's obviously not ideal.

**Comment by hsanson** [comment, marked as answer]
> I can't run from the project root (assuming I started vim from the project root) because the isort fixer changes the CWD to the dir of the file.

If you remove the cwd command does it work for you?

```
    return {
    \   'cwd': '%s:h',                                                                  <---- Remove all this line
    \   'command': ale#Escape(l:executable) . l:exec_args
    \   . (!empty(l:options) ? ' ' . l:options : '') . ' -',
    \}
```

If it does you may want to make a PR with the change.
